### DISCUSSION TITLE
Convert RGBM png to 32bit tga?

### DISCUSSION THREAD

**Discussion by Igameart** [ImageMagick/ImageMagick#2235] [Help]
Pretty self explanatory.

I have some PNG images in RGBM 8888 format that I'm exporting from a game engine. I'm looking to convert them into 32bit TGA. Is this possible with imagemagick?

**Comment by snibgo** [comment]
What version of IM, on what platform?

RGBM is mentioned on https://en.wikipedia.org/wiki/RGBE_image_format . Is that what you have?

You say your files are PNG format. Are they? Please link to a sample file.

**Comment by Igameart** [comment]
I don't have any samples that I can share, but it's just a standard png. RGBM is a method for compressing HDR imagery into a standard RGBA8 formatted image. 

You can see more about that here:
http://graphicrants.blogspot.com/2009/04/rgbm-color-encoding.html

**Comment by snibgo** [comment]
Going by the description of RGBM in http://graphicrants.blogspot.com/2009/04/rgbm-color-encoding.html, and assuming color and rgbm there are in the nominal range [0,1], something like this should work, where in.png is a 8888 RGBM image and out.tga is a conventional TGA image. Windows BAT syntax.
```
magick ^
  in.png ^
  -channel RGBA ^
  -evaluate Multiply 255/QuantumLevel ^
  +channel ^
  ( +clone ^
    -alpha extract ^
  ) ^
  -alpha off ^
  -compose Multiply -composite ^
  -evaluate 6 ^
  out.tga
```

**Comment by Igameart** [comment]
This looks great, but I'm having some errors with the evaluate function:
`invalid argument for option '-evaluate': 255/QuantumLevel`

as well as with the second evaluate:
`unrecognized evaluate operator 6`

I have changed the code to this:

```
  -channel RGBA ^
  -evaluate Multiply 255/8 ^
  ( +clone ^
    -alpha extract ^
  ) ^
  -alpha off ^
  -compose Multiply -composite ^
  -evaluate Multiply 6 ^
```

However it's outputting 8bit TGAs, and I need them to be 32bit. I've even added `-depth 32` and they are still output as 8 bit TGAs.

**Comment by snibgo** [comment]
What version of IM do you use, on what platform? I assume v7 with HDRI on Windows BAT.

Sorry, my fault, should be:
```
 -evaluate Multiply %%[fx:255/QuantumRange] ^
```
... and ...
```
 -evaluate Multiply 6 ^
```
I expect the output has 8 bits/channel/pixel. 3 channels would be 24 bits total per pixel. If you want 32 bits, that would be 4 channels, so add "-alpha set" before writing the output.

```
%IMG7%magick ^
  toes.png ^
  -channel RGBA ^
  -evaluate Multiply %%[fx:255/QuantumRange] ^
  +channel ^
  ( +clone ^
    -alpha extract ^
  ) ^
  -alpha off ^
  -compose Multiply -composite ^
  -evaluate Multiply 6 ^
  -alpha set ^
  out.tga
```

**Comment by Igameart** [comment]
I'm using version 7 with HDRI I do believe. I'm still getting an error with the first evaluate:
`%[fx:255/QuantumRange]`
What is quantum range? Is this bit depth of the source image?

**Comment by snibgo** [comment]
> I'm still getting an error ...

If you say what the error message is, that would help diagnosis.

QuantumRange is 255 if you use Q8, 65535 if you use Q16, 4294967295 for Q32 and 18446744073709551615 for Q64.

**Comment by Igameart** [comment]
It doesn't give me any more information than I've given you unfortunately. I removed the reference to quantum range since it's irrelevant information in this case, since it would just be 255/255.

Now it's producing some results. I'm going to run some more tests and try to ensure that the conversion is working accurately.
